rcl/include/rcl/log_level.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__LOG_LEVEL_H_
+#define RCL__LOG_LEVEL_H_
+
+#include "rcl/allocator.h"
+#include "rcl/macros.h"
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/// A logger item to specify a name and log_level
+typedef struct rcl_logger_setting_t
+{
+  /// name for the logger.
+  char * name;
+  /// level for the logger.
+  int level;
+} rcl_logger_setting_t;
+
+/// Hold log_level.
+typedef struct rcl_log_level_t
```

Comment:
Naming is confusing IMO.
It's not  clear that `rcl_log_level_t` is an array of `rcl_logger_setting_t` (the later has a `level` member, which makes things more confusing).

Maybe:

`rcl_logger_setting_t` -> `rcl_logger_setting_t`
`rcl_log_level_t` -> `rcl_logger_setting_array_t`

or

`rcl_logger_setting_t`
`rcl_log_level_t` -> `rcl_loggers_settings_t`

I'm not sure what is the right naming

Reply:
actually I like to use rcl_log_level_t, because user use '--log-level' to control it

Reply:
How about updating with
```
/// A logger item to specify a name and a log level
typedef struct rcl_logger_setting_t
{
  /// name for the logger.
  char * name;
  /// level for the logger.
  rcl_log_severity_t level;
} rcl_logger_setting_t;

/// Hold default logger level and other logger setting.
typedef struct rcl_log_level_t
{
  /// Default logger level
  rcl_log_severity_t default_logger_level;
  /// Array of logger setting
  struct rcl_logger_setting_t * logger_settings;
  /// Number of logger settings
  size_t num_logger_settings;
  rcl_allocator_t allocator;
} rcl_log_level_t;

typedef struct rcl_arguments_impl_t {
    rcl_log_level_t * log_level_settings;
}
```

Reply:
`rcl_log_level_t` sounds like a single log level, i.e. it sounds like a typedef of https://github.com/ros2/rcutils/blob/049167533d473cca5c8ae8347becdd2209503927/include/rcutils/logging.h#L165.

It should be clear in the name that it's a collection of logger settings IMO.

Reply:
`rcl_log_level_t` -> `rcl_log_levels_t`
works for me

Reply:
Updated